light/txpool.go

```diff
@@ -278,7 +278,7 @@ func (pool *TxPool) setNewHead(ctx context.Context, newHeader *types.Header) (tx
 	if idx := newHeader.Number.Uint64(); idx > pool.clearIdx+txPermanent {
 		idx2 := idx - txPermanent
 		for i := pool.clearIdx; i < idx2; i++ {
-			hash := core.GetCanonicalHash(pool.chainDb, i)
+			hash := pool.chain.GetStoredCanonicalHash(i)
```

Comment:
No, it is a single atomic db read. We never used locks before for accessing a single canonical hash, the only reason I added a lock is because I wanted to access multiple elements of the canonical chain and be sure that they belonged to the same version of the chain. The reason I'm using this function in the tx pool is that it is cached (unlike core.GetCanonicalHash).